### PR TITLE
CPS-126/CPB-219: DDL and private schema.js support

### PR DESCRIPTION
**Pull request by Johnbastian**
- `type` property is now included with every operation, either being `public`, `ddl` or `private`
- `ddl` operations are now detected by the operation’s (programmatic) name, i.e. directory, if ending with `_ddl`
  - if no schema is provided, a default schema is set
  - if a schema is provided, then this is used
- `private` type operations are never added to connectors.json, but are useful for required field validation and disabling globals

The rules for operation types and being added to the connectors.json op list is as follows:
- If no schema is present
  - If op's name ends with `DDL`, then it is added to connectors.json as a DDL type op
  - Else operation is NOT added to connectors.json
- If a schema is present
  - If type is NOT provided, then assume public by default (unless the name ends with `_ddl`)
  - If type is provided
    - If `public`, add to connectors.json
    - If `ddl`, add to connectors.json
    - If `private`, NOT added to connectors.json

### CODE REVIEW

**Comment by Johnbastian** [comment]
This PR covers tickets:
CPB-219
CPB-230

**Comment by Johnbastian** [comment]
Note: this update to falafel also needs an update to Connector Builder, to support this type property for DDLs.

File: test/buildConnectorsJson/prettyTitle.js [hunk 1]
```diff
@@ -5,46 +5,55 @@ var prettyTitle = require('../../lib/buildConnectorsJson/prettyTitle');
...
+		assert.strictEqual(prettyTitle('lurl'), 'Lurl');
+		assert.strictEqual(prettyTitle('list baseurl'), 'List baseurl');
+	});
+
+	it('should upper case ddl', function () {
+		assert.strictEqual(prettyTitle('ddl'), 'DDL');
+		assert.strictEqual(prettyTitle('list_ddl'), 'List DDL');
+
+		assert.strictEqual(prettyTitle('lurl'), 'Lurl');
+		assert.strictEqual(prettyTitle('list baseurl'), 'List baseurl');
```
Comment (jdon) [inline test/buildConnectorsJson/prettyTitle.js:48]: I don't think these should be in that test. Looks like your copied too much from the above test  😀

Comment (Johnbastian) [reply test/buildConnectorsJson/prettyTitle.js]: good catch

**Pull request merged**